### PR TITLE
Notes for Differential Drive and Gazebo

### PR DESCRIPTION
**Pull request by stevendes**
My notes for Diff-Drive and Gazebo

### CODE REVIEW

File: milner_desvars/diff-drive.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# how a differential-drive robot works and how to control it
+
+A differential-drive robot is a wheeled robot with two controllable wheels, this wheels only can turn by their axis, so rotation of the robot is accomplished by rotating the wheels in opposite directions.
```
Comment (eborghi10) [inline milner_desvars/diff-drive.md]: **these** wheels.

- this: singular
- these: plural

File: milner_desvars/diff-drive.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# how a differential-drive robot works and how to control it
```
Comment (eborghi10) [inline milner_desvars/diff-drive.md:1]: Please add more information about the unicycle model and how that relates to the [Twist](https://docs.ros.org/api/geometry_msgs/html/msg/Twist.html) message.

Comment (eborghi10) [reply milner_desvars/diff-drive.md:1]: That information can be related to what you already wrote.

File: milner_desvars/gazebo.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
+#Notes
+
+##Client and server separation
+
+When you run Gazebo you are actually running to different executables. In one hand "gzserver" and in the other "gzclient".
```
Comment (eborghi10) [inline milner_desvars/gazebo.md]: You can use `this Markdown style` instead for `gzserver` and `gzclient`.

In order to do that, use this:

```
`this is a one-line code`
```

File: milner_desvars/gazebo.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+The gzserver executable runs the physics update-loop and sensor data generation. This is the core of Gazebo and it can be used independtly of a graphical interphase.
```
Comment (eborghi10) [inline milner_desvars/gazebo.md]: Typo: independently

Comment (eborghi10) [reply milner_desvars/gazebo.md]: Typo: interface
Check for what "GUI" stands for on Google.

File: milner_desvars/gazebo.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
+#Notes
+
+##Client and server separation
+
+When you run Gazebo you are actually running to different executables. In one hand "gzserver" and in the other "gzclient".
+
+The gzserver executable runs the physics update-loop and sensor data generation. This is the core of Gazebo and it can be used independtly of a graphical interphase. 
+
+The gzclient executable runs a QT based user interface. This application provides a nice visualization of simulation, and convenient controls over various simulation properties.
```
Comment (eborghi10) [inline milner_desvars/gazebo.md]: Is it possible to run `gzserver` alone without `gzclient` ?

File: milner_desvars/gazebo.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+## World files
```
Comment (eborghi10) [inline milner_desvars/gazebo.md:13]: Could you check at least one example of these in the repo?

Comment (eborghi10) [reply milner_desvars/gazebo.md:13]: For example, an example of a world file, model file, and plugin. Don't worry if you don't find a plugin.

File: milner_desvars/gazebo.md [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+
+## Model files
+
+A model file uses the same SDF format as world files, but should only contain a single <model> ... <model>.The purpose of these files is to facilitate model reuse, and simplify world files.
+
+## Enviroment Variables
+
+Gazebo uses a number of environment variables to locate files, and set up communications between the server and clients. Default values that work for most cases are compiled in. Some enviroment variables are:
+
+GAZEBO_MODE_PATH
```
Comment (eborghi10) [inline milner_desvars/gazebo.md]: Use lists in Markdown and code for the names. More info [here](https://en.support.wordpress.com/markdown-quick-reference/).

File: milner_desvars/gazebo.md [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+GAZEBO_RESOURCE_PATH
+GAZEBO_MASTER_URI
+GAZEBO_PLUGIN_PATH
+GAZEBO_MODEL_DATABASE_URI
+
+## Plugins
+
+Plugins provide a simple and convenient mechanism to interface with Gazebo. Plugins can either be loaded on the command line, or specified in an SDF file.
+
+Plugins specified on the command line are loaded fist, then plugins specified in the SDF files are loaded. Some plugins are loaded by the server, such as plugins which affect physics properties, while other plugins are loaded by the graphical client to facillitate custom GUI generation.
```
Comment (eborghi10) [inline milner_desvars/gazebo.md]: Typo: facilitate

File: milner_desvars/gazebo.md [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+
+## Communication Betwen Processes
+
+Gzserver and Gzclient communicate using the gazebo communication library.
+
+The communication library currently uses the open source Google Protobuf for the message serialization and boost::ASIO for the transport mechanism. It supports the publish/subscribe communcation paradigm.
```
Comment (eborghi10) [inline milner_desvars/gazebo.md]: And what's the difference with ROS?

File: milner_desvars/gazebo.md [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+
+## System
+
+### Gazebo Master
+
+This is essentially a topic name server. It provides name lookup, and topic management. A single master can handle multiple physics simulations, sensor generators, and GUIs.
+
+### Communication Library
+
+Used by almost all subsequent libraries. Acts as the communication and transport mechanism for Gazebo
```
Comment (eborghi10) [inline milner_desvars/gazebo.md]: Missing dot at the end.

File: milner_desvars/gazebo.md [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+
+### Physics Library
+
+Provides a simple and generic interface to fundamental simulation components, such as rigid bodies, collision shapes, jounts, etc...
```
Comment (eborghi10) [inline milner_desvars/gazebo.md]: Typo: joints.

File: milner_desvars/gazebo.md [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+
+### Rendering Library
+
+Uses OGRE to provide a somple interface for rendering 3D scenes to both the GUI and sensor libraries.
```
Comment (eborghi10) [inline milner_desvars/gazebo.md]: Typo: simple

File: milner_desvars/gazebo.md [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+
+### Sensor Generation
+
+Implements all the varoius types of sensors, listens to the world state updates from a physics simulator and produces output specified by the instantianed sensors.
```
Comment (eborghi10) [inline milner_desvars/gazebo.md]: Typo: various

Comment (eborghi10) [reply milner_desvars/gazebo.md]: Typo: instantiated

File: milner_desvars/gazebo.md [hunk 14]
```diff
@@ -0,0 +1,71 @@
...
+
+### GUI 
+
+The GUI library uses QT to create graphical widgets for usters to interact with the simulation. The user may control the flow of time by pausing or changing time step size via GUI widgeds
```
Comment (eborghi10) [inline milner_desvars/gazebo.md]: Typo: users

Comment (eborghi10) [reply milner_desvars/gazebo.md]: Missing final dot.

**Review by eborghi10 — CHANGES_REQUESTED** [review]
First pass done, PTAL.

I recommend you to install Grammarly, a Chrome extension that checks grammatical errors in English.

File: milner_desvars/diff-drive.md [hunk 15]
```diff
@@ -0,0 +1,5 @@
...
+
+A differential-drive robot is a wheeled robot with two controllable wheels, this wheels only can turn by their axis, so rotation of the robot is accomplished by rotating the wheels in opposite directions.
+
+You can control the robot by controlling the velocity and direction in which the left and right wheels turn, you can use control theory to move the robot by giving the general velocity and direction you want it to move, and by that you calculate the velocities of both wheels to accomplish that objective.
```
Comment (ernestmc) [inline milner_desvars/diff-drive.md]: The information is not enough to actually control a real robot. Please investigate about how to model the movement and the equations used to obtain the differential speed of each wheel based on the desired linear and angular speeds for the robot.

File: milner_desvars/gazebo.md [hunk 16]
```diff
@@ -0,0 +1,71 @@
+#Notes
```
Comment (ernestmc) [inline milner_desvars/gazebo.md]: leave a space between the `#` symbol and the text for the title to be correctly interpreted

File: milner_desvars/gazebo.md [hunk 17]
```diff
@@ -0,0 +1,71 @@
...
+
+##Client and server separation
```
Comment (ernestmc) [inline milner_desvars/gazebo.md]: Same comment as above

File: milner_desvars/gazebo.md [hunk 18]
```diff
@@ -0,0 +1,71 @@
...
+
+When you run Gazebo you are actually running to different executables. In one hand "gzserver" and in the other "gzclient".
+
+The gzserver executable runs the physics update-loop and sensor data generation. This is the core of Gazebo and it can be used independtly of a graphical interphase. 
+
+The gzclient executable runs a QT based user interface. This application provides a nice visualization of simulation, and convenient controls over various simulation properties.
+
+## World files
+
+Contains the elements of the simulation, robots, sensors, etc. This file is formatted using SDF (Simation Description Format), and typically have a .world extension.
```
Comment (ernestmc) [inline milner_desvars/gazebo.md]: Simation --> simulation?

File: milner_desvars/gazebo.md [hunk 19]
```diff
@@ -0,0 +1,71 @@
...
+
+The Gazebo server reads this file to generate and populate a world.
+
+## Model files
+
+A model file uses the same SDF format as world files, but should only contain a single <model> ... <model>.The purpose of these files is to facilitate model reuse, and simplify world files.
```
Comment (ernestmc) [inline milner_desvars/gazebo.md]: What is this: `<model> ... <model>`

File: milner_desvars/gazebo.md (hunk 9 above)
Comment (ernestmc) [reply milner_desvars/gazebo.md]: communcation --> communication

**Review by ernestmc — COMMENTED** [review]
Left some comments. The Gazebo part is very good but the diff-drive part needs more work to be something useful.

**Review by eborghi10 — APPROVED** [review]

**Review by ernestmc — APPROVED** [review]
lgtm

**Comment by ernestmc** [comment]
@stevendes please squash this PR so we can merge it.

**Pull request closed without merging**